sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/ReactorNettyHttpResponseBase.java

```diff
@@ -31,8 +43,167 @@ public final String getHeaderValue(String name) {
 
     @Override
     public final HttpHeaders getHeaders() {
-        HttpHeaders headers = new HttpHeaders();
-        reactorNettyResponse.responseHeaders().forEach(e -> headers.put(e.getKey(), e.getValue()));
+        if (headers == null) {
+            headers = new NettyHttpHeaders(reactorNettyResponse.responseHeaders());
+        }
         return headers;
     }
+
+    // This class wraps a Netty HttpHeaders instance and provides an azure-core HttpHeaders view onto it.
+    // This avoids the need to copy the Netty HttpHeaders into an azure-core HttpHeaders instance.
+    static class NettyHttpHeaders extends HttpHeaders {
```

Comment:
Can we extract this into its own package-private class

Reply:
I've no aversion to classes like this and generally use this approach all the time - it keeps the package cleaner too.